var/spack/repos/builtin/packages/autoconf/package.py

```diff
@@ -55,12 +55,11 @@ def filter_sbang(self):
         # We have to do this after install because otherwise the install
         # target will try to rebuild the binaries (filter_file updates the
         # timestamps)
-        perl = join_path(self.spec['perl'].prefix.bin, 'perl')
 
         # Revert sbang, so Spack's sbang hook can fix it up
         filter_file('^#! /usr/bin/env perl',
-                    '#! {0} -w'.format(perl),
-                    '{0}/autom4te'.format(self.prefix.bin),
+                    '#! {0} -w'.format(self.spec['perl'].command.path),
+                    join_path(self.prefix.bin, 'autom4te'),
```

Comment:
Another newish feature you might not know, you can use:
```python
self.prefix.bin.autom4te
```
instead of:
```python
join_path(self.prefix.bin, 'autom4te')
```
I personally prefer the former, but I added this feature, so I'm biased.

Reply:
Fixed, thanks! I actually saw (and liked) that in some other package but was not sure whether it worked for arbitrary commands. :-)

Reply:
It works for any path as long as it is a valid attribute name in Python. So if it starts with a number or contains a dash, it might not work, but it should work for everything else.